cmake/Modules/FindMKL.cmake

```diff
@@ -88,25 +88,41 @@ ELSE ("${SIZE_OF_VOIDP}" EQUAL 8)
   SET(iccvers "ia32")
   SET(mkl64s)
 ENDIF ("${SIZE_OF_VOIDP}" EQUAL 8)
-IF(CMAKE_COMPILER_IS_GNUCC)
-  IF ("${MKL_THREADING}" STREQUAL "TBB")
-    SET(mklthreads "mkl_tbb_thread")
-    SET(mklrtls "tbb")
-  ELSE()
-    SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
-    SET(mklrtls "gomp" "iomp5")
-  ENDIF()
-  SET(mklifaces  "intel" "gf")
-ELSE(CMAKE_COMPILER_IS_GNUCC)
+
+IF(WIN32)
   IF ("${MKL_THREADING}" STREQUAL "TBB")
     SET(mklthreads "mkl_tbb_thread")
-    SET(mklrtls "tbb")
+    IF (CMAKE_BUILD_TYPE STREQUAL Debug)
+      SET(mklrtls "tbb12_debug")
+    ELSE ()
+      SET(mklrtls "tbb12")
+    ENDIF ()
   ELSE()
     SET(mklthreads "mkl_intel_thread")
-    SET(mklrtls "iomp5" "guide")
+    SET(mklrtls "libiomp5md")
   ENDIF()
-  SET(mklifaces  "intel")
-ENDIF (CMAKE_COMPILER_IS_GNUCC)
+    SET(mklifaces  "intel")
+ELSE(WIN32)
+  IF(CMAKE_COMPILER_IS_GNUCC)
+    IF ("${MKL_THREADING}" STREQUAL "TBB")
+      SET(mklthreads "mkl_tbb_thread")
+      SET(mklrtls "tbb")
+    ELSE()
+      SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
```

Comment:
It looks like this PR changes preference from Intel's OpenMP library to GNU one. I'm fine with it, but would prefer to do it in separate change

Reply:
No, I didn't change any thing for original code. 
<img width="617" alt="image" src="https://github.com/user-attachments/assets/bd5a1394-58a3-4d45-bf2d-2806dd6a1997">

I just move the original code to `else` block of `Win32`.

Reply:
Looks like Github change highlight is sometimes off. Here is the correct change view:
<img width="1415" alt="Screenshot 2024-07-15 at 9 22 18 AM" src="https://github.com/user-attachments/assets/00b3359a-d4c3-42fa-877d-bcbdd9e9257b">